lib/js/sportsday.js

```diff
@@ -253,6 +244,8 @@
     $scope.time = 120;
     $scope.plural = "s";
     $scope.$on("$routeChangeStart", function() {
+      $scope.time = 120;
```

Comment:
This resets the timer every time the route changes. When the route changes, data is reloaded anyway - Angular**JS** doesn't come with an automatic cache unlike Angular.io and therefore data is reloaded on page changes. This timer is simply to reload data when the user is idle on a page but is still looking at it, without navigating to other pages.

Reply:
BTW: a future solution for real-time data is https://firebase.google.com/products/firestore/, a NoSQL database **that can be called straight from AngularJS** without needing a backend of any sort. On their free plan, you can store 1 GiB data and have 50k reads/day. It's automatically real-time - whenever you query data, it binds it to an object that gets zero-latency updates via a [WebSocket](https://en.wikipedia.org/wiki/WebSocket), meaning you don't need to refresh the page. It's the same technology real-time chat systems like Slack use. Firestore has really simple code too, and database queries are still only one line long, even if you want them to be bound to a real-time object.